server_settings.txt
/etc/nginx/sites-available/geekshop


server {
    listen 80;
    server_name 89.108.77.224;

    location = /favicon.ico { access_log off; log_not_found off; }
    location /static/ {
            root /home/django/7-Django_Base_geekshop;
    }
    location /media/ {
            root /home/django/7-Django_Base_geekshop;
    }

    location / {
            include proxy_params;
            proxy_pass http://unix:/home/django/7-Django_Base_geekshop/geekshop.sock;
    }
 }




/etc/systemd/system/gunicorn.service


[Unit]
Description=gunicorn daemon
After=network.target

[Service]
User=django
Group=www-data
WorkingDirectory=/home/django/7-Django_Base_geekshop
ExecStart=/home/django/env/bin/gunicorn --access-logfile - --workers 3 --bind unix:/home/django/7-Django_Base_geekshop/geekshop.sock geekshop.wsgi

[Install]
WantedBy=multi-user.target







history.txt
    1  exit
    2  apt update
    3  apt upgrade
    4  apt install nginx
    5  apt install postgresql postgresql-contrib 
    6  apt install python3-venv
    7  apt install git
    8  nano /etc/postgresql/12/main/pg_hba.conf 
    9  systermctl restart postgresql
   10  systemctl restart postgresql
   11  systemctl status postgresql
   12  useradd -g www-data -m django
   13  cd /home/django/
   14  python3 -m venv env
   15  ssh-keygen
   16  cd ..
   17  cat /root/.ssh/id_rsa.pub 
   18  cd /home/django/
   19  ls
   20  git clone git@github.com:morozov-git/7-Django_Base_geekshop.git
   21  ls
   22  source env/bin/activate
   23  cd 7-Django_Base_geekshop/
   24  cd geekshop/
   25  git checkout Django_p2_lesson5
   26  ls
   27  ls -li
   28  cd ..
   29  ls -li
   30  pip install -r requirements.txt 
   31  pip install bdist_wheel
   32  pip install wheel
   33  pip install -r requirements.txt 
   34  pip install gunicorn
   35  psql -U postgres
   36  cd geekshop/
   37  psql -U postgres
   38  cd ..
   39  systemctl status postgresql
   40  psql -U postgres
   41  psql
   42  sudo -u postgresql psql
   43  sudo -U postgres psql
   44  sudo -u postgres psql
   45  psql -U postgres
   46  cd ..
   47  nano etc/postgresql/12/main/pg_hba.conf 
   48  cd /home/django/7-Django_Base_geekshop/
   49  ls
   50  psql -U postgres
   51  systemctl restart postgresql
   52  systemctl status postgresql
   53  psql -U postgres
   54  python3 manage.py migrate
   55  cd geekshop/
   56  python3 manage.py migrate
   57  cd ..
   58  systemctl status postgresql
   59  python3 manage.py migrate
   60  psql -U postgres
   61  python3 manage.py migrate
   62  python3.8 manage.py migrate
   63  python3 manage.py migrate
   64  pip install -r requirements.txt 
   65  python3 manage.py migrate
   66  cd ..
   67  python3 manage.py migrate
   68  cd 7
   69  cd 7-Django_Base_geekshop/
   70  git pull
   71  git branch
   72  ls -li
   73  psql -U postgres
   74  python3 manage.py migrate
   75  cd /home/django/
   76  cd 7-Django_Base_geekshop/
   77  git pull
   78  python3 manage.py migrate
   79  psql -U postgres
   80  source env/bin/activate
   81  cd ..
   82  source env/bin/activate
   83  cd 7-Django_Base_geekshop/
   84  python3 manage.py migrate
   85  python3 manage.py loaddata db.json 
   86  python3 manage.py fill_db
   87  python3 manage.py createsuperuser
   88  python3 manage.py runserver 0.0.0.0:8000
   89  python3 manage.py fill_db
   90  python3 manage.py runserver 0.0.0.0:8000
   91  git pull
   92  python3 manage.py runserver 0.0.0.0:8000
   93  ls -li
   94  nano .env
   95  nano .env.simple 
   96  nano .env
   97  python3 manage.py runserver 0.0.0.0:8000
   98  nano .env
   99  python3 manage.py runserver 0.0.0.0:8000
  100  git pull
  101  python3 manage.py runserver 0.0.0.0:8000
  102  git pull
  103  python3 manage.py runserver 0.0.0.0:8000
  104  git pull
  105  python3 manage.py runserver 0.0.0.0:8000
  106  git pull
  107  python3 manage.py makemigrations
  108  git pull
  109  python3 manage.py makemigrations
  110  git pull
  111  python3 manage.py makemigrations
  112  git pull
  113  python3 manage.py makemigrations
  114  git pull
  115  python3 manage.py makemigrations
  116  python3 manage.py migrate
  117  psql -U postgres
  118  python3 manage.py makemigrations
  119  python3 manage.py migrate
  120  python3 manage.py createsuperuser
  121  python3 manage.py fill_db
  122  python3 manage.py runserver 0.0.0.0:8000
  123  python3 manage.py fill_db
  124  git pull
  125  python3 manage.py makemigrations
  126  python3 manage.py migrate
  127  python3 manage.py fill_db
  128  python3 manage.py runserver 0.0.0.0:8000
  129  git pull
  130  python3 manage.py makemigrations
  131  python3 manage.py migrate
  132  python3 manage.py runserver 0.0.0.0:8000
  133  nano /etc/systemd/system/gunicorn.service
  134  systemctl enable gunicorn
  135  systemctl start gunicorn
  136  systemctl status gunicorn
  137  ls
  138  cd ..
  139  ls
  140  nano /etc/systemd/system/gunicorn.service
  141  systemctl enable gunicorn
  142  systemctl start gunicorn
  143  systemctl status gunicorn
  144  systemctl disable gunicorn
  145  systemctl enable gunicorn
  146  systemctl start gunicorn
  147  systemctl status gunicorn
  148  systemctl start gunicorn
  149  systemctl status gunicorn
  150  nano /etc/systemd/system/gunicorn.service
  151  systemctl disable gunicorn
  152  systemctl enable gunicorn
  153  systemctl start gunicorn
  154  systemctl status gunicorn
  155  ls
  156  cd env/bin/
  157  ls
  158  cd gunicorn
  159  cd gunicorncd ..
  160  cd ..
  161  ls
  162  cd 7§
  163  cd 7-Django_Base_geekshop/
  164  ls
  165  cd geekshop/
  166  ls
  167  cd ..
  168  nano /etc/systemd/system/gunicorn.service
  169  systemctl disable gunicorn
  170  systemctl enable gunicorn
  171  systemctl start gunicorn
  172  systemctl status gunicorn
  173  nano /etc/systemd/system/gunicorn.service
  174  systemctl disable gunicorn
  175  systemctl enable gunicorn
  176  systemctl start gunicorn
  177  systemctl status gunicorn
  178  nano /etc/systemd/system/gunicorn.service
  179  ls
  180  cd 7-Django_Base_geekshop/
  181  nano /etc/systemd/system/gunicorn.service
  182  systemctl disable gunicorn
  183  systemctl enable gunicorn
  184  systemctl start gunicorn
  185  systemctl status gunicorn
  186  cd ..
  187  cd django/
  188  ls
  189  cd env/
  190  cd bin/
  191  ls
  192  cat gunicorn 
  193  nano /etc/systemd/system/gunicorn.service
  194  systemctl disable gunicorn
  195  systemctl enable gunicorn
  196  systemctl start gunicorn
  197  systemctl status gunicorn
  198  systemctl start gunicorn
  199  systemctl status gunicorn
  200  nano /etc/nginx/sites-available/geekshop
  201  cd ..
  202  ls
  203  cd 7-Django_Base_geekshop/
  204  ls
  205  ln -s /etc/nginx/sites-available/geekshop /etc/nginx/sites-enabled/
  206  rm /etc/nginx/sites-enabled/default 
  207  nginx -t

geekshop/settings.py
"""
Django settings for geekshop project.

Generated by 'django-admin startproject' using Django 3.2.6.

For more information on this file, see
https://docs.djangoproject.com/en/3.2/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.2/ref/settings/
"""
from typing import Any, Union

from dotenv import load_dotenv
from pathlib import Path
import os

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR: Union[Path, Any] = Path(__file__).resolve().parent.parent
# BASE_DIR = os.path.dirname(os.path.abspath(__file__))

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.2/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-i!_k%(enhzl3x8r7rrxvuie(m06n(_u&*kk5j5-gkbmhl!+%+8'

load_dotenv(BASE_DIR / '.env')

# SECURITY WARNING: don't run  with debug turned on in production!
DEBUG = True
# DEBUG = False

ALLOWED_HOSTS = ['*']

# Application definition

INSTALLED_APPS = [
	'django.contrib.admin',
	'django.contrib.auth',
	'django.contrib.contenttypes',
	'django.contrib.sessions',
	'django.contrib.messages',
	'django.contrib.staticfiles',

	#   our app's
	'products',
	'users',
	'baskets',
	'admins',
	'social_django',
	'ordersapp',
	'debug_toolbar',
	'template_profiler_panel',
	'django_extensions',

]

MIDDLEWARE = [
	# 'django.middleware.cache.UpdateCacheMiddleware',
	'django.middleware.security.SecurityMiddleware',
	'django.contrib.sessions.middleware.SessionMiddleware',
	'django.middleware.common.CommonMiddleware',
	# 'django.middleware.csrf.CsrfViewMiddleware',   # отключать перед использованием тестов
	'django.contrib.auth.middleware.AuthenticationMiddleware',
	'django.contrib.messages.middleware.MessageMiddleware',
	'django.middleware.clickjacking.XFrameOptionsMiddleware',
	'social_django.middleware.SocialAuthExceptionMiddleware',
	'debug_toolbar.middleware.DebugToolbarMiddleware',
	# 'django.middleware.cache.FetchFromCacheMiddleware',

]

ROOT_URLCONF = 'geekshop.urls'

TEMPLATES = [
	{
		'BACKEND': 'django.template.backends.django.DjangoTemplates',
		'DIRS': [BASE_DIR / 'templates']
		,
		'APP_DIRS': True,
		'OPTIONS': {
			'context_processors': [
				'django.template.context_processors.debug',
				'django.template.context_processors.request',
				'django.contrib.auth.context_processors.auth',
				'django.contrib.messages.context_processors.messages',
				'geekshop.context_processors.basket',
				'social_django.context_processors.backends',
				'social_django.context_processors.login_redirect',
			],
		},
	},
]

WSGI_APPLICATION = 'geekshop.wsgi.application'

# Database
# https://docs.djangoproject.com/en/3.2/ref/settings/#databases

DATABASES = {
	'default': {
		'ENGINE': 'django.db.backends.sqlite3',
		'NAME': os.path.join(BASE_DIR / 'db.sqlite3'),
		# 'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
	}
}

# DATABASES = {
# 	'default': {
# 		'ENGINE': 'django.db.backends.postgresql',
# 		'NAME': 'geekshop',
# 		'USER': 'postgres',
# 	}
# }

# Password validation
# https://docs.djangoproject.com/en/3.2/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
	{
		'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
	},
	{
		'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
	},
	{
		'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
	},
	{
		'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
	},
]

# Internationalization
# https://docs.djangoproject.com/en/3.2/topics/i18n/

# LANGUAGE_CODE = 'en-us'
LANGUAGE_CODE = 'ru-ru'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.2/howto/static-files/

STATIC_URL = '/static/'

STATICFILES_DIRS = (BASE_DIR / 'static',)
# STATIC_ROOT = os.path.join(BASE_DIR, 'static')

# STATICFILES_DIRS = (
# 	os.path.join(BASE_DIR, 'assets'),
# )
STATICFILES_FINDERS = (
	'django.contrib.staticfiles.finders.FileSystemFinder',
	'django.contrib.staticfiles.finders.AppDirectoriesFinder',
)
# Default primary key field type
# https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

MEDIA_URL = '/media/'
MEDIA_ROOT = BASE_DIR / 'media'
AUTH_USER_MODEL = 'users.User'
LOGIN_URL = '/users/login/'

DOMAIN_NAME = 'http://localhost:8000'
EMAIL_HOST = 'localhost'
EMAIL_HOST_USER = os.getenv('EMAIL_HOST_USER')
EMAIL_HOST_PASSWORD = os.getenv('EMAIL_HOST_PASSWORD')
EMAIL_USE_SSL = True if os.getenv('EMAIL_USE_SSL') == 'True' else False

# настройки для EmailBackend сервера
EMAIL_PORT = '25'
EMAIL_USE_SSL = True if os.getenv('EMAIL_USE_SSL') == 'True' else False
EMAIL_BACKEND = 'django.core.mail.backends.filebased.EmailBackend'
EMAIL_FILE_PATH = 'tmp/emails/'

# настройки для локального почтового сервера
# EMAIL_PORT = '25'
# EMAIL_USE_SSL = False
# EMAIL_HOST_USER, EMAIL_HOST_PASSWORD = None, None
# python -m smtpd -n -c DebuggingServer localhost:25

AUTHENTICATION_BACKENDS = (
	'django.contrib.auth.backends.ModelBackend',
	'social_core.backends.vk.VKOAuth2'
)

SOCIAL_AUTH_VK_OAUTH2_KEY = os.getenv('SOCIAL_AUTH_VK_OAUTH2_KEY')
SOCIAL_AUTH_VK_OAUTH2_SECRET = os.getenv('SOCIAL_AUTH_VK_OAUTH2_SECRET')
SOCIAL_AUTH_VK_OAUTH2_API_VERSION = os.getenv('SOCIAL_AUTH_VK_OAUTH2_API_VERSION')
LOGIN_ERROR_URL = '/'
SOCIAL_AUTH_VK_OAUTH2_IGNORE_DEFAULT_SCORE = True
SOCIAL_AUTH_VK_OAUTH2_SCORE = ['email']
SOCIAL_AUTH_URL_NAMESPACE = 'social'

# SOCIAL_AUTH_REDIRECT_IS_HTTPS = True  # проверить авторизацию через другие аккаунты
# SOCIAL_AUTH_JSONFIELD_ENABLED = True
# SESSION_COOKIE_SECURE = False

SOCIAL_AUTH_PIPELINE = (
	'social_core.pipeline.social_auth.social_details',
	'social_core.pipeline.social_auth.social_uid',
	'social_core.pipeline.social_auth.auth_allowed',
	'social_core.pipeline.social_auth.social_user',
	'social_core.pipeline.user.create_user',
	'users.pipeline.save_user_profile',
	'social_core.pipeline.social_auth.associate_user',
	'social_core.pipeline.social_auth.load_extra_data',
	'social_core.pipeline.user.user_details',
)

# if DEBUG:
# 	def show_toolbar(request):
# 		return True
#
#
# 	DEBUG_TOOLBAR_CONFIG = {
# 		'SHOW_TOOLBAR_CALLBACK': show_toolbar,
# 	}
#
# 	DEBUG_TOOLBAR_PANELS = [
# 		'debug_toolbar.panels.versions.VersionsPanel',
# 		'debug_toolbar.panels.timer.TimerPanel',
# 		'debug_toolbar.panels.settings.SettingsPanel',
# 		'debug_toolbar.panels.headers.HeadersPanel',
# 		'debug_toolbar.panels.request.RequestPanel',
# 		'debug_toolbar.panels.sql.SQLPanel',
# 		'debug_toolbar.panels.templates.TemplatesPanel',
# 		'debug_toolbar.panels.staticfiles.StaticFilesPanel',
# 		'debug_toolbar.panels.cache.CachePanel',
# 		'debug_toolbar.panels.signals.SignalsPanel',
# 		'debug_toolbar.panels.logging.LoggingPanel',
# 		'debug_toolbar.panels.redirects.RedirectsPanel',
# 		'debug_toolbar.panels.profiling.ProfilingPanel',
# 		'template_profiler_panel.panels.template.TemplateProfilerPanel',
# 	]

# if os.name == 'posix':
CACHE_MIDDLEWARE_ALIAS = 'default'
CACHE_MIDDLEWARE_SECONDS = 120
CACHE_MIDDLEWARE_KEY_PREFIX = 'geekshop'
CACHES = {'default':
			  {'BACKEND': 'django.core.cache.backends.memcached.MemcachedCache',
			   'LOCATION': '127.0.0.1:11211', }
		  }
LOW_CACHE = True
